src/commands/run.rs

```diff
@@ -113,8 +114,35 @@ impl RunCommand {
         }
 
         // Load the main wasm module.
-        self.handle_module(&store, &module_registry)
-            .with_context(|| format!("failed to run main module `{}`", self.module.display()))?;
+        match self
+            .handle_module(&store, &module_registry)
+            .with_context(|| format!("failed to run main module `{}`", self.module.display()))
+        {
+            Ok(()) => (),
+            Err(e) => {
+                // If the program exited because of a trap, return an error code
+                // to the outside environment indicating a more severe problem
+                // than a simple failure.
+                if let Some(source) = e.source() {
+                    if let Some(source) = source.source() {
+                        if source.is::<Trap>() {
```

Comment:
Rather than hard-coding "check for a trap two error sources above this error in the chain" I think it would make sense to loop over the whole chain and check for if any of them is a trap. Does that make sense?

Reply:
```rust
let mut err = Some(&e);
while let Some(source) = err {
    if source.is::<Trap>() { ... }
    err = source.source();
}
```

Reply:
Makes sense to me! I've now added a patch to do that. I just had to make a slight tweek, since `e` here is an `anyhow::Error`, so we have to start with the first `source()` rather than starting with `e`.